app/assets/javascripts/components/MentionInput.jsx

```diff
@@ -0,0 +1,126 @@
+
+class MentionInput extends React.Component {
+
+    constructor(props) {
+      super();
+      this.state = {
+        users: [],
+        originalUsers: [],
+        idToUsers: {},
+        showUsers: false,
+        mentionedUsers: [],
+        value: ''
+      }
+    }
+
+    componentDidMount() {
+      Requester.get('/api/users/show').then((data) => {
+        this.setState({ users: data });
+        this.setState({ originalUsers: Array.from(data) })
+        let userIdToName = {};
+        data.forEach((user) => {
+            userIdToName[user.id] = user.first_name + " " + user.last_name;
+        });
+        this.setState({ idToUsers: userIdToName });
+      });
+    }
+    
+    userClicked = (evt) => {
+      const cursor = this.textInput.selectionStart;
+      const curr = this.textInput.value;
+      const closestAt = curr.lastIndexOf('@', cursor);
```

Comment:
This is also assuming that people always mention at the end of the message i think?

Reply:
yeah it does -- i'm honestly not sure how to make it so if someone sticks a random @ in the middle of a sentence that it still works. slack doesn't handle that case either (just puts the @ in the text) so i think we're good?